envoy/stream_info/boolean_accessor.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/stream_info/filter_state.h"
+
+namespace Envoy {
+namespace StreamInfo {
+
+/**
+ * A FilterState object that tracks a single boolean value.
+ */
+class BooleanAccessor : public FilterState::Object {
+public:
+  /**
+   * @return the tracked value.
+   */
+  virtual bool value() const PURE;
```

Comment:
I think in the past we used a string accessor with "reason" as value. That might be more informative than a boolean.

Reply:
I thought about it being confusing to users, but I guess that it works if it's a pattern already in use. I deleted the boolean accessor and used string accessor instead.